pilot/cmd/pilot-agent/main.go

```diff
@@ -638,8 +643,8 @@ func init() {
 		"Connection timeout used by Envoy for supporting services")
 	proxyCmd.PersistentFlags().StringVar(&statsdUDPAddress, "statsdUdpAddress", values.StatsdUdpAddress,
 		"IP Address and Port of a statsd UDP listener (e.g. 10.75.241.127:9125)")
-	proxyCmd.PersistentFlags().StringVar(&envoyMetricsServiceAddress, "envoyMetricsServiceAddress", values.EnvoyMetricsService.Address,
-		"Host and Port of an Envoy Metrics Service API implementation (e.g. metrics-service:15000)")
+	proxyCmd.PersistentFlags().StringVar(&envoyMetricsService, "envoyMetricsService", "",
```

Comment:
This can cause compatibility issues during upgrade

Reply:
Please see this https://github.com/istio/istio/pull/17046#discussion_r323648002 - I removed based on this.

Reply:
We'd better just keep this(mark deprecated and not use it) instead of letting user report bugs.

Reply:
@rshriram what do you think? Should I retain this?

Reply:
How will this cause compatibility issues ? We can mark it as deprecated if it does.

Reply:
> We'd better just keep this(mark deprecated and not use it) instead of letting user report bugs.

@hzxuzhonghu  We have deprecated this at API , are you suggesting we leave this arg and not process it?

Reply:
we do allow unknown flags now https://github.com/istio/istio/blob/master/pilot/cmd/pilot-agent/main.go#L131. So should not be an issue.